includes/documents/abstract-wcpdf-order-document.php

```diff
@@ -224,6 +232,59 @@ public function save_settings( $latest = false ) {
 			}
 		}
 	}
+	
+	public function init_number( $force_new_number = false ) {
+		$lock            = new Semaphore( $this->lock_name, $this->lock_time, $this->lock_loggers, $this->lock_context );
+		$document_number = $this->exists() ? $this->get_data( 'number' ) : null;
+		$document_number = ! empty( $document_number ) && $force_new_number ? null : $document_number;
+		
+		if ( $lock->lock( $this->lock_retries ) && empty( $document_number ) ) {
+			
+			$lock->log( "Lock acquired for the {$this->slug} number init.", 'info' );
+			
+			try {
+				// If a third-party plugin claims to generate document numbers, trigger this instead
+				if ( apply_filters( "woocommerce_{$this->slug}_number_by_plugin", false ) || apply_filters( "wpo_wcpdf_external_{$this->slug}_number_enabled", false, $this ) ) {
+					$document_number = apply_filters( "woocommerce_generate_{$this->slug}_number", $document_number, $this->order );  // legacy (backwards compatibility)
+					$document_number = apply_filters( "woocommerce_{$this->slug}_number", $document_number, $this->order->get_id() ); // legacy (backwards compatibility)
+					$document_number = apply_filters( "wpo_wcpdf_external_{$this->slug}_number", $document_number, $this );
+				} elseif ( isset( $this->settings['display_number'] ) && 'order_number' === $this->settings['display_number'] && ! empty( $this->order ) ) {
+					$document_number = $this->order->get_order_number();
```

Comment:
Do you think we can change this part?

Order numbers are always present in the Woo database, and we don't need to save them in our table. So we can always have the document separate numbering system if no third party changes it.

We can easily display the order number if the user wants that as the document number in the method related to display.

Reply:
This will involve more work that could bring more issues, because there is a class for the numbers to apply formatting, and I want to avoid too much drastic changes. I think we should keep this simple for now.